File: sdk/python/kucoin_universal_sdk/generate/account/fee/api_fee_test.py
import unittest
from .model_get_basic_fee_req import GetBasicFeeReq
from .model_get_basic_fee_resp import GetBasicFeeResp
from .model_get_futures_actual_fee_req import GetFuturesActualFeeReq
from .model_get_futures_actual_fee_resp import GetFuturesActualFeeResp
from .model_get_spot_actual_fee_req import GetSpotActualFeeReq
from .model_get_spot_actual_fee_resp import GetSpotActualFeeResp
from kucoin_universal_sdk.model.common import RestResponse


class FeeAPITest(unittest.TestCase):

    def test_get_basic_fee_req_model(self):
        """
       get_basic_fee
       Get Basic Fee - Spot/Margin
       /api/v1/base-fee
       """
        data = "{\"currencyType\": 1}"
        req = GetBasicFeeReq.from_json(data)

    def test_get_basic_fee_resp_model(self):
        """
        get_basic_fee
        Get Basic Fee - Spot/Margin
        /api/v1/base-fee
        """
        data = "{\n    \"code\": \"200000\",\n    \"data\": {\n        \"takerFeeRate\": \"0.001\",\n        \"makerFeeRate\": \"0.001\"\n    }\n}"
        common_response = RestResponse.from_json(data)
        resp = GetBasicFeeResp.from_dict(common_response.data)

    def test_get_spot_actual_fee_req_model(self):
        """
       get_spot_actual_fee
       Get Actual Fee - Spot/Margin
       /api/v1/trade-fees
       """
        data = "{\"symbols\": \"BTC-USDT,ETH-USDT\"}"
        req = GetSpotActualFeeReq.from_json(data)

    def test_get_spot_actual_fee_resp_model(self):
        """
        get_spot_actual_fee
        Get Actual Fee - Spot/Margin
        /api/v1/trade-fees
        """
        data = "{\"code\":\"200000\",\"data\":[{\"symbol\":\"BTC-USDT\",\"takerFeeRate\":\"0.001\",\"makerFeeRate\":\"0.001\"},{\"symbol\":\"ETH-USDT\",\"takerFeeRate\":\"0.001\",\"makerFeeRate\":\"0.001\"}]}"
        common_response = RestResponse.from_json(data)
        resp = GetSpotActualFeeResp.from_dict(common_response.data)

    def test_get_futures_actual_fee_req_model(self):
        """
       get_futures_actual_fee
       Get Actual Fee - Futures
       /api/v1/trade-fees
       """
        data = "{\"symbol\": \"XBTUSDTM\"}"
        req = GetFuturesActualFeeReq.from_json(data)

    def test_get_futures_actual_fee_resp_model(self):
        """
        get_futures_actual_fee
        Get Actual Fee - Futures
        /api/v1/trade-fees
        """
        data = "{\"code\":\"200000\",\"data\":{\"symbol\":\"XBTUSDTM\",\"takerFeeRate\":\"0.0006\",\"makerFeeRate\":\"0.0002\"}}"
        common_response = RestResponse.from_json(data)
        resp = GetFuturesActualFeeResp.from_dict(common_response.data)

File: sdk/golang/pkg/generate/account/fee/api_fee_test.go
package fee

import (
	"encoding/json"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/types"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFeeGetBasicFeeReqModel(t *testing.T) {
	// GetBasicFee
	// Get Basic Fee - Spot/Margin
	// /api/v1/base-fee

	data := "{\"currencyType\": 1}"
	req := &GetBasicFeeReq{}
	err := json.Unmarshal([]byte(data), req)
	req.ToMap()
	assert.Nil(t, err)
}

func TestFeeGetBasicFeeRespModel(t *testing.T) {
	// GetBasicFee
	// Get Basic Fee - Spot/Margin
	// /api/v1/base-fee

	data := "{\n    \"code\": \"200000\",\n    \"data\": {\n        \"takerFeeRate\": \"0.001\",\n        \"makerFeeRate\": \"0.001\"\n    }\n}"
	commonResp := &types.RestResponse{}
	err := json.Unmarshal([]byte(data), commonResp)
	assert.Nil(t, err)
	assert.NotNil(t, commonResp.Data)
	resp := &GetBasicFeeResp{}
	err = json.Unmarshal([]byte(commonResp.Data), resp)
	resp.ToMap()
	assert.Nil(t, err)
}

func TestFeeGetSpotActualFeeReqModel(t *testing.T) {
	// GetSpotActualFee
	// Get Actual Fee - Spot/Margin
	// /api/v1/trade-fees

	data := "{\"symbols\": \"BTC-USDT,ETH-USDT\"}"
	req := &GetSpotActualFeeReq{}
	err := json.Unmarshal([]byte(data), req)
	req.ToMap()
	assert.Nil(t, err)
}

func TestFeeGetSpotActualFeeRespModel(t *testing.T) {
	// GetSpotActualFee
	// Get Actual Fee - Spot/Margin
	// /api/v1/trade-fees

	data := "{\"code\":\"200000\",\"data\":[{\"symbol\":\"BTC-USDT\",\"takerFeeRate\":\"0.001\",\"makerFeeRate\":\"0.001\"},{\"symbol\":\"ETH-USDT\",\"takerFeeRate\":\"0.001\",\"makerFeeRate\":\"0.001\"}]}"
	commonResp := &types.RestResponse{}
	err := json.Unmarshal([]byte(data), commonResp)
	assert.Nil(t, err)
	assert.NotNil(t, commonResp.Data)
	resp := &GetSpotActualFeeResp{}
	err = json.Unmarshal([]byte(commonResp.Data), resp)
	resp.ToMap()
	assert.Nil(t, err)
}

func TestFeeGetFuturesActualFeeReqModel(t *testing.T) {
	// GetFuturesActualFee
	// Get Actual Fee - Futures
	// /api/v1/trade-fees

	data := "{\"symbol\": \"XBTUSDTM\"}"
	req := &GetFuturesActualFeeReq{}
	err := json.Unmarshal([]byte(data), req)
	req.ToMap()
	assert.Nil(t, err)
}

func TestFeeGetFuturesActualFeeRespModel(t *testing.T) {
	// GetFuturesActualFee
	// Get Actual Fee - Futures
	// /api/v1/trade-fees

	data := "{\"code\":\"200000\",\"data\":{\"symbol\":\"XBTUSDTM\",\"takerFeeRate\":\"0.0006\",\"makerFeeRate\":\"0.0002\"}}"
	commonResp := &types.RestResponse{}
	err := json.Unmarshal([]byte(data), commonResp)
	assert.Nil(t, err)
	assert.NotNil(t, commonResp.Data)
	resp := &GetFuturesActualFeeResp{}
	err = json.Unmarshal([]byte(commonResp.Data), resp)
	resp.ToMap()
	assert.Nil(t, err)
}
